Initialize ScrollBehavior's content size to infinite rather than 0.

mpcomplete opened the pull request:
This way, we don't clamp the scroll offset until after the first layout,
when we update the ScrollBehavior's contentSize to its actual value.
This only matters when restoring a Scrollable to a previous
scrollOffset. Since it was valid before the Scrollable was disposed, we
assume its still valid when restored.

BUG=https://github.com/flutter/flutter/issues/4883
BUG=https://github.com/flutter/flutter/issues/4797

abarth commented:
Test?

HansMuller commented on packages/flutter/lib/src/widgets/scroll_behavior.dart:
```diff
@@ -225,7 +225,7 @@ class OverscrollBehavior extends BoundedBehavior {
 class OverscrollWhenScrollableBehavior extends OverscrollBehavior {
   /// Creates a scroll behavior that allows overscrolling only when some amount of scrolling is already possible.
-  OverscrollWhenScrollableBehavior({ double contentExtent: 0.0, double containerExtent: 0.0, double minScrollOffset: 0.0 })
+  OverscrollWhenScrollableBehavior({ double contentExtent: double.INFINITY, double containerExtent: 0.0, double minScrollOffset: 0.0 })
```
Per our discussion: this change should be applied to all of the scroll behaviors, since ScrollConfiguration produces OverscrollWhenScrollableBehavior by default but not unconditionally.

For the benefit of someone reading this code years from now, it might be good to explain the rationale for the double.INFINITY. We don't want to cause a scrollable's initial scroll offset, to be inadvertently clipped before a valid contentExtent is computed.

HansMuller commented:
This is a deft little fix. Definitely needs a regression test.

LGTM

mpcomplete commented:
OK, all ScrollBehaviors now initialize to INFINITE contentExtent. And found a good place to add the regression test.

mpcomplete committed:
Initialize ScrollBehavior's content size to infinite rather than 0.

Why this matters: If you navigate back to a page with a Scrollable that
has a nonzero scrollOffset, we will restore that scrollOffset. We clamp
the scrollOffset to the contentExtent before the first layout, before
contentExtent is updated to its proper value. Initializing contentExtent
to INFINITY effectively disables the first clamp, until we can get a
valid value from layout. Since the previous scrollOffset was valid, it
seems safe to assume it's still valid.

BUG=https://github.com/flutter/flutter/issues/4883
BUG=https://github.com/flutter/flutter/issues/4797
